gene/pointMutation.go

```diff
@@ -0,0 +1,129 @@
+package gene
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/numbers"
+)
+
+// PointMutation changes a single nucleotide to the desired base, predicts the effect,
+// and updates the Gene struct to reflect the change.
+// The position of the mutation should be given in base-zero genomic coordinates.
+func PointMutation(g *Gene, genomePos int, alt dna.Base) (EffectPrediction, error) {
+	var answer EffectPrediction
+	answer.StopDist = -1
+	var err error
+
+	err = pointPreRunChecks(g, genomePos, &alt)
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	genomeIndexPos := numbers.AbsInt(genomePos - g.startPos) // abs needed to handle negative posStrand
+	g.genomeSeq[genomeIndexPos] = alt
+
+	cdnaIndexPos := int(g.featureArray[genomeIndexPos])
+
+	if cdnaIndexPos >= 0 { // mutation is coding
+		err = pointCoding(g, cdnaIndexPos, alt, &answer)
+	} else { // mutation is noncoding
+		err = pointNonCoding(g, genomePos, &answer)
+	}
+
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	g.protSeq = dna.TranslateSeq(g.codingSeq.seq) // TODO improve efficiency by updating the protein as changes are made
+	return answer, nil
+}
+
+// checkSplice inputs a cDNA offset value and determines if a variant at this site may affect splicing.
+func checkSplice(CdnaOffset int) MutationType {
+	if numbers.AbsInt(CdnaOffset) <= 2 {
+		return Splice
+	}
+
+	if numbers.AbsInt(CdnaOffset) <= 10 {
+		return FarSplice
+	}
+
+	return Intronic
```

Comment:
It seems like all we know is that it is not close to a splice site.  Maybe it could be coding or in a utr?  Not sure why it would need to be an intron.